src/System.Private.CoreLib/shared/System/Text/Unicode/Utf8Utility.WhiteSpace.cs

```diff
@@ -0,0 +1,141 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using Internal.Runtime.CompilerServices;
+
+#pragma warning disable SA1121 // explicitly using type aliases instead of built-in types
+#if BIT64
+using nint = System.Int64;
+using nuint = System.UInt64;
+#else
+using nint = System.Int32;
+using nuint = System.UInt32;
+#endif
+
+namespace System.Text.Unicode
+{
+    internal static partial class Utf8Utility
+    {
+        /// <summary>
+        /// Returns the index in <paramref name="utf8Data"/> where the first non-whitespace character
+        /// appears, or the input length if the data contains only whitespace characters.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int GetIndexOfFirstNonWhiteSpaceChar(ReadOnlySpan<byte> utf8Data)
+        {
+            return (int)GetIndexOfFirstNonWhiteSpaceChar(ref MemoryMarshal.GetReference(utf8Data), (uint)utf8Data.Length);
```

Comment:
```suggestion
            return (int)GetIndexOfFirstNonWhiteSpaceChar(ref MemoryMarshal.GetReference(utf8Data), (nuint)utf8Data.Length);
```
As the method's parameter is `nunit`?
The same for `GetIndexOfTrailingWhiteSpaceSequence`.

Reply:
This may apply to GetIndexOfTrailingWhiteSpaceSequence too.

---
In reply to: [324546438](https://github.com/dotnet/coreclr/pull/26711#discussion_r324546438) [](ancestors = 324546438)

Reply:
The cast to `uint` instead of `nuint` is intentional. It allows the JIT to emit a zero-latency `mov` instead of a sign-extending `movsxd`.

Reply:
Ah, it's the clever `(ulong)(uint)value`-trick you applied in https://github.com/dotnet/coreclr/pull/20771
Nice 👍